### DISCUSSION TITLE
Sunbird-Ed / SunbirdEd-mobile-app - Trailing query string in the content path

### DISCUSSION THREAD

**Discussion by chandu4002** [Sunbird-Ed/Community#478] [Bugs] [answered]
Issue Description:
When we are accessing the following assessment, it is loading upto 75% and then screen is freezing over there. In the logs we are seeing the following url along with 404 not Found.
For the logs PFA screenshots.

contentID: do_31374720144239001618
Unable to access the following url:
https://obj.diksha.gov.in/ntp-content-production/content/ecml/do_31374720144239001618-latest/widgets/content-plugins/org.ekstep.questionset.quiz-1.0/manifest.json?BUILD_NUMBER

In the above url, at the end  [BUILD_NUMBER](https://obj.diksha.gov.in/ntp-content-production/content/ecml/do_31374720144239001618-latest/widgets/content-plugins/org.ekstep.questionset.quiz-1.0/manifest.json?BUILD_NUMBER) is getting appended to the url. When we remove that string we able to access the url.

About the same we checked with Oracle Team, As per them it is the issue in sunbird code regarding same I have attached one mail screenshot for your reference.

@ravishtrigyn @swayangjit @vinukumar-vs @rajeshkumaravel @anandvarada

<img width="960" alt="Screenshot 2023-04-28 181913" src="https://user-images.githubusercontent.com/129141695/235842706-d82645b3-e2ac-406b-a97d-f504de39795a.png">
<img width="960" alt="Screenshot 2023-04-28 182006" src="https://user-images.githubusercontent.com/129141695/235842712-b5569139-1284-4e02-910a-a44fa7344de0.png">
<img width="960" alt="image-20230427-054609" src="https://user-images.githubusercontent.com/129141695/235842713-0fb46271-808b-4c68-ae61-d8767d6d5aec.png">
<img width="209" alt="Screenshot 2023-04-28 181801" src="https://user-images.githubusercontent.com/129141695/235842717-5f2cb107-6f83-4f97-8cc4-d2cf8c83d451.png">

**Comment by Abhign** [comment]
@swayangjit Can you check this issue.

CC: @Krishnaj20 @chitranshu-keshav @vinukumar-vs

**Comment by swayangjit** [comment]
@chandu4002 It seems "obj.diksha.in" infra doesn't support query params. In mobile app all the assets are accessed via BUILD_NUMBER. for cache-busting purpose. Please ask devops team to support query params in the URLs or URLs with only query/key  like following
https://obj.diksha.gov.in/ntp-content-production/content/ecml/do_31374720144239001618-latest/widgets/content-plugins/org.ekstep.questionset.quiz-1.0/manifest.json?BUILD_NUMBER
.

**Reply by vinukumar-vs** [reply, marked as answer]
from Portal the API requests are going as below & it is working fine build query params with BUILD_NUMBER & without BUILD_NUMBER.

https://diksha.gov.in/content-plugins/org.ekstep.questionunit.mcq-1.2/renderer/template/template_controller.js??BUILD_NUMBER

https://diksha.gov.in/content-plugins/org.ekstep.questionunit.mcq-1.2/renderer/template/template_controller.js??BUILD_NUMBER&_=1683107652169

Similarly please make sure files can be accessible for mobile with & without BUILD_NUMBER

https://obj.diksha.gov.in/ntp-content-production/content/ecml/do_31374720144239001618-latest/widgets/content-plugins/org.ekstep.questionset.quiz-1.0/manifest.json?BUILD_NUMBER

https://obj.diksha.gov.in/ntp-content-production/content/ecml/do_31374720144239001618-latest/widgets/content-plugins/org.ekstep.questionset.quiz-1.0/manifest.json?BUILD_NUMBER=1683107652169
